src/app/api/cronjob/route.ts

```diff
@@ -0,0 +1,177 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import { addDays, eachDayOfInterval, format, isWithinInterval } from "date-fns";
+import { and, eq, gte, lte } from "drizzle-orm";
+
+import { db } from "@/lib/db";
+import {
+  gameSessions,
+  gameSessionSchedules,
+  semesters,
+  weekdayEnum,
+} from "@/lib/db/schema";
+import { getWeekday } from "@/lib/utils";
+import { insertGameSessionSchema } from "@/lib/validators";
+import { Weekday } from "@/types/types";
+
+export async function POST(req: NextRequest) {
+  try {
+    // get current day, correct to NZ timezone and reset to 12am
+    let now: Date = new Date();
+    now = new Date(now.getTime() - now.getTimezoneOffset() * 60000);
+    now = new Date(now.getTime() - (now.getTime() % (3600 * 24 * 1000)));
+
+    // Testing date
+    //now = new Date(2024, 1, 21);
+    //now = new Date(2024, 1, 22);
+    //now = new Date(2024, 2, 28);
+    //now = new Date(2024, 2, 29);
+    //now = new Date(2024, 3, 7);
+    //now = new Date(2024, 3, 8);
+    //now = new Date(2024,5,24);
+    //now = new Date(2024,5,25);
+    //now = new Date(now.getTime() - now.getTimezoneOffset() * 60000);
+    //console.log(now);
+
+    const currentDateUTC = new Date(
+      now.getTime() + now.getTimezoneOffset() * 60000,
+    );
+
+    // Find current semester
+    const activeSemester = await db.query.semesters.findFirst({
+      where: and(
+        lte(
+          semesters.startDate,
+          format(addDays(currentDateUTC, 7), "yyyy-MM-dd"),
+        ),
+        gte(semesters.endDate, format(currentDateUTC, "yyyy-MM-dd")),
+      ),
+    });
+
+    if (!activeSemester) {
+      return NextResponse.json("No active semester found.", { status: 400 });
+    }
+
+    //const semesterStartDate = new Date(activeSemester.startDate);
+    const semesterEndDate = new Date(activeSemester.endDate);
+    const semesterBreakStart = new Date(activeSemester.breakStart);
+    const semesterBreakEnd = new Date(activeSemester.breakEnd);
+    const bookingOpenDay = activeSemester.bookingOpenDay;
+    const diff = findDiff(now, bookingOpenDay);
+    const startDate = addDays(currentDateUTC, diff);
+
```

Comment:
have a look at the functions provided by datefns https://date-fns.org/v3.6.0/docs/
you could simplify this as `parse(bookingOpenDay, "iiii", currentDateUTC)`

although you might have to check that `isAfter(parsedDate, currentDateUTC)`